test/config/test/browser-configs.test.ts

```diff
@@ -303,3 +306,143 @@ test('can enable browser-cli options for multi-project workspace', async () => {
   expect(projects[1].config.browser.enabled).toBe(true)
   expect(projects[1].config.browser.headless).toBe(true)
 })
+
+function getCliConfig(options: UserConfig, cli: string[], fs: Record<string, string> = {}) {
+  const root = resolve(process.cwd(), `vitest-test-${crypto.randomUUID()}`)
+  useFS(root, {
+    ...fs,
+    'basic.test.ts': /* ts */`
+      import { test } from 'vitest'
+      test('basic', () => {
+        expect(1).toBe(1)
+      })
+    `,
+    'vitest.config.ts': /* ts */ `
+      export default {
+        test: {
+          reporters: [
+            {
+              onInit(vitest) {
+                const browser = vitest.config.browser
+                console.log(JSON.stringify({
+                  browser: {
+                    headless: browser.headless,
+                    browser: browser.enabled,
+                    ui: browser.ui,
+                  },
+                  workspace: vitest.projects.map(p => {
+                    return {
+                      name: p.name,
+                      headless: p.config.browser.headless,
+                      browser: p.config.browser.enabled,
+                      ui: p.config.browser.ui,
+                    }
+                  })
+                }))
+                // throw an error to avoid running tests
+                throw new Error('stop')
```

Comment:
This utility looks great - I've thought about building something like this before. Sometimes it's not enough to check resolved options just via `resolveConfig` as plugins and other parts can change them before tests run. Bailing out like this before pools are running saves time.